app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/RadioGroup.kt

```diff
@@ -0,0 +1,905 @@
+            lockedState = lockedState,
+          )
+        }
+      }
+    }
+  }
+}
+
+@Composable
+private fun HorizontalWithLabelRadioOption(
+  optionText: String,
+  onClick: () -> Unit,
+  lockedState: LockedState,
+  chosenState: ChosenState,
+  enabled: Boolean,
+  style: TextStyle,
+  textColor: Color,
+  modifier: Modifier = Modifier,
+) {
+  val interactionSource = remember { MutableInteractionSource() }
+  Row(
+    verticalAlignment = Alignment.CenterVertically,
+    modifier = modifier
+      .clickable(
+        enabled = enabled,
+        role = Role.RadioButton,
+        interactionSource = interactionSource,
+        indication = ripple(
+          bounded = false,
+          radius = 50.dp,
+          // todo: pls check out this ripple. I like this one, but maybe it's too much
+        ),
+      ) {
+        if (enabled) {
+          onClick()
+        }
+      },
+  ) {
+    SelectIndicationCircle(
+      chosenState,
+      lockedState,
+    )
+    Spacer(Modifier.width(8.dp))
+    HedvigText(
+      optionText,
+      style = style,
+      color = textColor,
+      modifier = Modifier,
+    )
+  }
+}
+
+@Composable
+private fun VerticalRadioGroup(
+  data: List<OptionData>,
+  radioGroupStyle: Vertical,
+  groupLockedState: LockedState,
+  radioGroupSize: RadioGroupSize,
+  onOptionClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  Column(modifier) {
+    for (radioOptionData in data) {
+      val radioOptionStyle = when (radioGroupStyle) {
+        is Vertical.Default -> Default
+        is Vertical.Icon -> RadioOptionStyle.Icon(radioOptionData.iconResource ?: Painter(R.drawable.pillow_hedvig)) // todo: how is that for placeholder
+        is Vertical.Label -> Label(radioOptionData.labelText ?: "") // todo: no placeholder
+        is Vertical.LeftAligned -> LeftAligned
+      }
+      RadioOption(
+        data = radioOptionData,
+        radioOptionStyle = radioOptionStyle,
+        groupLockedState = groupLockedState,
+        radioOptionSize = radioGroupSize.toOptionSize(),
+        onOptionClick = {
+          onOptionClick(radioOptionData.id)
+        },
+      )
+      Spacer(Modifier.height(4.dp))
+    }
+  }
+}
+
+@Composable
+private fun VerticalRadioGroupWithLabel(
+  data: List<OptionData>,
+  radioGroupStyle: VerticalWithGroupLabel,
+  groupLockedState: LockedState,
+  radioGroupSize: RadioGroupSize,
+  contentPaddingValues: PaddingValues,
+  onOptionClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  Surface(
+    shape = radioGroupSize.getShape(),
+    color = radioOptionColors.containerColor,
+  ) {
+    Column(
+      modifier,
+    ) {
+      val labelTextColor = radioOptionColors.labelTextColor(groupLockedState)
+      HedvigText(
+        modifier = Modifier.padding(contentPaddingValues),
+        text = radioGroupStyle.groupLabelText,
+        style = MediumSizeRadioOptionTokens.LabelTextFont.value, // same for all sizes in figma
```

Comment:
I think if it happens not to be too much of a hassle to add it inside the RadioGroupSize object, it can be there so that if the impl changes in the future and we only go and edit the tokens we don't get behavior that we do not expect by having only the medium tokens drive the value of this text.